### PR TITLE
Improve support for non CoCo attestation services

### PR DESCRIPTION
**Pull request by sameo**
Although KBS only supports the CoCo attestation service, it is designed to support any external AS.

This PR brings no functional changes but rearrange and rename code and features to more cleanly support external ASes.
As a consequence, the CoCo AS support is now moved under `api/src/attestation/coco`

### CODE REVIEW

File: src/api/src/attestation/mod.rs
Comment (Xynnn007) [inline src/api/src/attestation/mod.rs:39]: An idea hits me: as CoCo-AS and Amber can both support remote connection, we could integration both of the client code and let the configuration to determine the runtime behavior (which one to use) rather than compiling time. In this way an user would not need to worry about the complex different compile features? 

This way the binary would not be much bigger because only client code are included.

How do you think of this?

File: src/api/Cargo.toml
Comment (Xynnn007) [inline src/api/Cargo.toml:13]: A question not related to this PR. Why do we need a "no-verifier" feature? If it is for test, we could implement a mock-ed object of trait `Attest`

cc @jialez0

**Review by Xynnn007 — APPROVED** [review]
LGTM. Thanks! Only some comments not related to this PR.

File: Makefile
Comment (sameo) [reply Makefile:18]: I was intentionally trying to keep the Makefile target short, and since builtin is the default I though we could keep it shorter by not adding `builtin`.
But I really don't have any strong preference and can add `builtin` for consistency sake, if you prefer it this way.

File: src/api/src/attestation/mod.rs
Comment (sameo) [reply src/api/src/attestation/mod.rs]: Thanks, fixed now.

File: src/api/src/attestation/mod.rs
Comment (sameo) [reply src/api/src/attestation/mod.rs:39]: I think this is a very good idea. Should we wait for the Amber PR to be merged and then do that change?

File: src/api/Cargo.toml
Comment (sameo) [reply src/api/Cargo.toml:13]: The `no-verifier` is really for testing purposes, so that one can build the project without pulling all the tdx/sgx dependencies.
I agree a mocked implementation of the Attest trait would make more sense, I like that.

File: src/api/src/attestation/mod.rs
Comment (Xynnn007) [reply src/api/src/attestation/mod.rs:39]: Great, agreed!

File: src/api/Cargo.toml
Comment (Xynnn007) [reply src/api/Cargo.toml:13]: Great. I'll work on this then

File: Makefile
Comment (Xynnn007) [reply Makefile:18]: I'm also not strong with one choice. Let me do some refactor after Amber is merged. As now "built-in" is clearly for CoCo-Attestation-Service, we might use a more short feature name `"kbs-builtin-openssl"` and `"kbs-builtin"`

**Comment by sameo** [comment]
@jialez0 Could you PTAL?

**Comment by jialez0** [comment]
> @jialez0 Could you PTAL?

OK, I'll look at this PR :-)

**Review by jialez0 — APPROVED** [review]
LGTM.

**Pull request merged**
